server/src/main/java/org/opensearch/indices/ShardLimitValidator.java

```diff
@@ -64,12 +65,23 @@ public class ShardLimitValidator {
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
+
+    public static final Setting<Boolean> SETTING_CLUSTER_IGNORE_HIDDEN_INDEXES = Setting.boolSetting(
+        "cluster.ignore_hidden_indexes",
+        true,
+        Setting.Property.Dynamic,
+        Setting.Property.NodeScope
+    );
+
     protected final AtomicInteger shardLimitPerNode = new AtomicInteger();
+    protected final AtomicBoolean ignoreHiddenIndexes = new AtomicBoolean();
```

Comment:
why protected? can we just use a `volatile` boolean

Reply:
+1 -- this can be private and volatile.

Reply:
ACK.